Packet++/src/RawPacket.cpp

```diff
@@ -132,14 +132,16 @@ void RawPacket::appendData(const uint8_t* dataToAppend, size_t dataToAppendLen)
 
 void RawPacket::insertData(int atIndex, const uint8_t* dataToInsert, size_t dataToInsertLen)
 {
-	int index = m_RawDataLen - 1;
-	while (index >= atIndex)
+	// memmove copies data as if there was an intermediate buffer inbetween - so it allows for copying processes on overlapping src/dest ptrs
+	// if insertData is called with atIndex == m_RawDataLen, then no data is being moved. The data of the raw packet is still extended by dataToInsertLen
+	memmove((uint8_t*)m_RawData + atIndex + dataToInsertLen, (uint8_t*)m_RawData + atIndex, m_RawDataLen - atIndex);

```

Comment:
How do you make sure data isn't being overrun?

Consider this example:

Existing packet raw data:  `[0x01, 0x01, 0x01, 0x01, 0x01, 0x01 (unused) (unused) (unused)]`
Data to be added in index 3 is: `[0x02, 0x02, 0x02]`

The result of `memmove` will be: `[0x01, 0x01, 0x01, 0x02, 0x02, 0x02 (unused) (unused) (unused)]`

Instead of the expected result which is: `[0x01, 0x01, 0x01, 0x02, 0x02, 0x02, 0x01, 0x01, 0x01]`

That is the purpose of the the `while` loop...